manticore/core/cpu/abstractcpu.py

```diff
@@ -479,89 +554,98 @@ def read_bytes(self, where, size):
 
     #######################################
     # Decoder
-    @abstractmethod
     def _wrap_operands(self, operands):
         '''
         Private method to decorate a capstone Operand to our needs. See Operand
         class
         '''
-        pass
+        raise NotImplementedError
 
     def decode_instruction(self, pc):
         '''
-        This will decode an instruction from memory pointed by `pc` and store
-        it in self.instruction.
+        This will decode an instruction from memory pointed by `pc`
 
         :param int pc: address of the instruction
         '''
-        # No dynamic code!!! #TODO!
-
-        if issymbolic(pc):
-            raise SymbolicPCException()
-
-        if not self.memory.access_ok(pc,'x'):
-            raise InvalidPCException(pc)
-
-        #Check if instruction was already decoded
+        #No dynamic code!!! #TODO! 
+        #Check if instruction was already decoded 
         self._instruction_cache = {}
         if pc in self._instruction_cache:
             logger.debug("Intruction cache hit at %x", pc)
             return self._instruction_cache[pc]
 
+        def _read_concrete_byte(address):
+            #This reads a byte from memory ignoring permissions
+            #and concretize it if symbolic
+            c = self.memory[address]
+            if issymbolic(c):
+                assert isinstance(c, BitVec) and  c.size == 8
+                if isinstance(c, Constant):
+                    c = chr(c.value)
+                else:
+                    logger.error('Concretize executable memory %r %r', c, text )
+                    raise ConcretizeMemory(self.memory, address = pc,
+                                            size = 8 * self.max_instr_width, 
+                                            policy = 'INSTRUCTION' )
+            assert isinstance(c, str)
```

Comment:
Ideally it can't be anything else. 
But it could be an Expression if the concretizing code above failed. 
If that's double checked and we are certain I may just remove this.